crates/swc_ecma_minifier/src/lib.rs

```diff
@@ -233,7 +233,13 @@ pub fn optimize(
         let _timer = timer!("mangle names");
         // TODO: base54.reset();
 
-        let preserved = idents_to_preserve(mangle.clone(), &n);
+        let mut preserved = idents_to_preserve(mangle.clone(), &n);
+
+        // https://tc39.es/ecma262/multipage/global-object.html#sec-value-properties-of-the-global-object-infinity
+        let unresolved_ctxt = SyntaxContext::empty().apply_mark(marks.unresolved_mark);
+        preserved.insert(("undefined".into(), unresolved_ctxt));
+        preserved.insert(("NaN".into(), unresolved_ctxt));
+        preserved.insert(("Infinity".into(), unresolved_ctxt));
```

Comment:
Why need this here?

Reply:
SWC will rename the variable even it's unresolved.
Cannot find the reason now.

Reply:
Can we move this list to https://github.com/swc-project/swc/blob/1aefb0a7ccca8ea15ebb135c28ce88a8973adbf9/crates/swc_ecma_transforms_base/src/rename/analyzer/scope.rs#L201-L216 ?

Reply:
I just attempted it. Passing the `unresolved_mark` deeply to this function requires modifications at multiple layers.

Reply:
Ah, I see